### PR TITLE
fix: the inconsistent order of input/output in range select

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Fix the inconsistent order of input/output in range select

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- If the result of the project plan happens to be the schema of the range plan, no project plan is required
- Add projection in range select to transform the order of the schema in the range plan to the order of the output
- Modify the `sin` function used in the test case as it has precision issues on m1

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

https://github.com/GreptimeTeam/greptimedb/issues/1662

### CODE REVIEW

File: src/query/src/range_select/plan.rs [hunk 1]
```diff
@@ -118,18 +127,63 @@ impl RangeSelect {
...
+                }
+            }
+            if project_map.len() == projection_expr.len() {
+                Some(project_map)
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```
Comment (waynexia) [inline src/query/src/range_select/plan.rs:164]: non-blocking: it would be better if we can simplify this `if`s...

**Review by waynexia — APPROVED** [review]

File: src/query/src/range_select/plan.rs (hunk 1 above)
Comment (killme2008) [reply src/query/src/range_select/plan.rs:164]: 👍

Comment (killme2008) [reply src/query/src/range_select/plan.rs:164]: @Taylor-lagrange Hi, will u do this refactor? If not, I'll merge this PR.

Comment (Taylor-lagrange) [reply src/query/src/range_select/plan.rs:164]: Give me some time to refactor this

Comment (Taylor-lagrange) [reply src/query/src/range_select/plan.rs]: Done @killme2008

**Pull request merged**